impurityModel/test/test_h0.py

```diff
@@ -0,0 +1,58 @@
+import os
+from glob import glob
+
+from impurityModel.ed.get_spectra import read_pickled_file
+
+DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+def test_read_all_h0_pickle_files():
+    for h0_filename in glob(os.path.join(DIR_PATH, "../../h0/h0*.pickle")):
+        h0 = read_pickled_file(h0_filename)
+
+        string = os.path.basename(h0_filename).split(".")[0].split("_")[-1].split("bath")[0]
+        nBaths_tot = sum([int(nbath) for nbath in string.split("p")])
+        # So far, in all the non-interacting Hamiltonians the angular momentum is equal to two
+        # for the correlated orbitals that are coupled to the bath states.
+        nBaths = {2: nBaths_tot}
+
+        sanity_check_non_interacting_hamiltonian(h0, nBaths)
+
+
+def sanity_check_non_interacting_hamiltonian(h0: dict[tuple, complex], nBaths: dict[int, int]):
+    """
+    Sanity check non-interacting Hamiltonian operator.
+
+    Parameters
+    -------
+    h0:
+        Non-interacting Hamiltonian operator.
+        Describes impurity orbitals and bath orbitals.
+        Each tuple key describes a process of two steps: annihilation followed by creation.
+        Each step is described by a tuple of the form:
+        (spin_orb, 'c') for creation, and
+        (spin_orb, 'a') for annihilation,
+        where spin_orb is a tuple of the form (l, s, m) or (l, b) or ((l_a, l_b), b).
+    nBaths:
+        angular momentum: number of bath states coupled to the correlated orbitals with this angular momentum.
+    """
+    for process, value in h0.items():
+        assert isinstance(value, complex)
+        assert len(process) == 2  # two events in non-interacting Hamiltonian
+        assert process[1][1] == "a"  # First event is annihilation
+        assert process[0][1] == "c"  # Second event is creation
+        for event in process[::-1]:
```

Comment:
no need to check events in any particular order.
```suggestion
        for event in process:
```

Reply:
The order does not matter in this for-loop. But when a process is used for real, i.e. applied on a product state, the order matters. 

The order is chosen such that $c_i^\dagger c_j \ket{\Psi}$ is implemented as process `((i, 'c'), (j, 'a'))` operating on the state $\ket{\Psi}$ (with `'c'` denoting creation  and `'a'` annihilation). So the right-most "event" is first applied, and finally the left-most "event", just like the  math expression. 
Relevant documentation e.g. https://github.com/JohanSchott/impurityModel/blob/master/impurityModel/ed/finite.py#L1320
https://github.com/JohanSchott/impurityModel/blob/master/impurityModel/ed/finite.py#L1395

And in the same way is the Coulomb interaction process between two electrons: 
$c_i^\dagger c_j^\dagger c_k c_l$ implemented as `((i, 'c'), (j, 'c'), (k, 'a'), (l, 'a'))`
https://github.com/JohanSchott/impurityModel/blob/master/impurityModel/ed/finite.py#L1322

Because of this I think it might be good to consistently loop over events "from right to left" in this repo.